exercises/queen-attack/example.exs

```diff
@@ -1,20 +1,24 @@
 defmodule Queens do
   @type t :: %Queens{white: {integer, integer}, black: {integer, integer}}
-  defstruct white: {0, 3}, black: {7, 3}
+  defstruct [:white, :black]
+  @board_range 0..7
 
   @doc """
   Creates a new set of Queens
   """
-  @spec new() :: Queens.t()
-  @spec new({integer, integer}, {integer, integer}) :: Queens.t()
-  def new(same, same), do: raise(ArgumentError)
+  @spec new(white: {integer, integer} | nil, black: {integer, integer} | nil) :: Queens.t()
+  def new(opts \\ [])
+
+  def new(white: same, black: same),
+    do: raise(ArgumentError)
```

Comment:
I don't think this will ever match, as the keywords will be collected into a single keyword list.

Reply:
Yeah it doesn't work, I will implement this check another way.